include/picongpu/particles/densityProfiles/profiles.def

```diff
@@ -21,7 +21,7 @@
 
 #include "picongpu/particles/densityProfiles/EveryNthCellImpl.def"
 #include "picongpu/particles/densityProfiles/FreeFormulaImpl.def"
-#include "picongpu/particles/densityProfiles/FromHDF5Impl.def"
+#include "picongpu/particles/densityProfiles/FromOpenPMDImpl.def"
```

Comment:
@psychocoderHPC how we do it now, the declaration in the `.def` file is always available, and the definition is only included when `ENABLE_OPENPMD` is true. Do we want to keep it this way, or to also conditionally enable the declaration? I am for the latter, as now it may be confusing for a user that the template parameters are not right, and that's why the implementation is not found (happened to me).

Reply:
def file should be not avaiable or empty (contains guards) if the dependency is missing.
Currently we guard the include but mybe it is better to have guards within the files.

Reply:
Okay. So I will add guards inside the `.def` file so that the declaration is only available when the dependency is there. I think this way is slightly better as the code including the file doesn't have to know about its internal dependencies.